### PR TITLE
update Trust API and connect to UI

### PR DESCRIPTION
**Pull request by pinzart90**
### Purpose
https://jira.autodesk.com/browse/DYN-4905

In order for serialization to work, the properties in PreferenceSettings need to be public. 
Having a TrustLocManager with similar APIs will cause confusion (for devs at least)
In this PR I removed the trust api manager and put all trust apis in PreferenceSettings

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section** 

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Menu/TrustedPathViewModel.cs [hunk 1]
```diff
@@ -45,21 +52,18 @@ public virtual void OnRequestShowFileDialog(object sender, TrustedPathEventArgs
...
         public DelegateCommand SaveSettingCommand { get; private set; }
 
-        public TrustedPathViewModel()
-        { 
-            InitializeTrustedLocations();
-            InitializeCommands();
-        }
-
         /// <summary>
         /// This constructor overload has been added for backwards comptability.
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/ViewModels/Menu/TrustedPathViewModel.cs:56]: comment seems to be copy paste error?

File: src/DynamoCoreWpf/ViewModels/Menu/TrustedPathViewModel.cs [hunk 2]
```diff
@@ -100,6 +104,18 @@ private void InsertPath()
...
                 return;
 
+            try
+            {
+                PathHelper.ValidateDirectory(args.Path);
+            }
+            catch(Exception ex)
+            {
+                this.logger?.LogError($"Failed to add trusted location ${args.Path} due to the following error: {ex.Message}");
+                string errorMessage = string.Format(Resources.PackageFolderNotAccessible, args.Path);
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/ViewModels/Menu/TrustedPathViewModel.cs:114]: could path be null?

File: src/DynamoCore/Configuration/PreferenceSettings.cs [hunk 3]
```diff
@@ -768,5 +751,81 @@ private static void MigrateStdLibTokenToBuiltInToken(PreferenceSettings settings
...
+            {
+                string location = PathHelper.ValidateDirectory(path);
+                var existing = TrustedLocations.FirstOrDefault(x => x.Equals(location));
+                if (string.IsNullOrEmpty(existing))
+                {
+                    trustedLocations.Add(location);
+                    return true;
+                }
+            }
+            catch { }
```
Comment (QilongTang) [inline src/DynamoCore/Configuration/PreferenceSettings.cs:773]: We may want to use Dynamo logger to log the exception here

File: src/DynamoCore/Configuration/PreferenceSettings.cs [hunk 4]
```diff
@@ -768,5 +751,81 @@ private static void MigrateStdLibTokenToBuiltInToken(PreferenceSettings settings
...
+        }
+
+        /// <summary>
+        /// Set trusted locations in the PreferenceSettings configuration.
+        /// </summary>
+        /// <param name="locs"></param>
+        internal void SetTrustedLocations(IEnumerable<string> locs)
+        {
+            trustedLocations.Clear();
+            foreach (var loc in locs) trustedLocations.Add(loc);
```
Comment (QilongTang) [inline src/DynamoCore/Configuration/PreferenceSettings.cs:799]: Maybe use `Enumerable.Union()` or any default API for simplicity?

File: src/DynamoCore/Configuration/PreferenceSettings.cs (hunk 3 above)
Comment (mjkkirschner) [reply src/DynamoCore/Configuration/PreferenceSettings.cs:773]: yeah agreed, I think the easiest way to do this is the implement ILogSource and then hook up the events in the DynamoModel right after PreferenceSetting is constructed.

File: src/DynamoCore/Configuration/PreferenceSettings.cs [hunk 5]
```diff
@@ -304,24 +305,14 @@ public bool DisableTrustWarnings
         }
 
         /// <summary>
-        /// Backing store for TrustedLocations
+        /// Default trusted locations. A fixed list of locations (This field should not be serializable or public)
         /// </summary>
-        private List<string> trustedLocations { get; set; } = new List<string>();
+        private List<string> defaultTrustedLocations { get; set; } = new List<string>();
```
Comment (mjkkirschner) [inline src/DynamoCore/Configuration/PreferenceSettings.cs:310]: I'm confused, what is this for? Is it only used during testing?

**Review by mjkkirschner — COMMENTED** [review]
It looks okay to me though I don't understand the use default trusted paths list

**Review by QilongTang — APPROVED** [review]
LGTM

File: src/DynamoCore/Configuration/PreferenceSettings.cs (hunk 5 above)
Comment (pinzart90) [reply src/DynamoCore/Configuration/PreferenceSettings.cs:310]: `defaultTrustedLocations` was supposed to be the default trusted paths list (like the programData/appdata dynamo packages folders)
But now that I look at some of the jira [comments](https://jira.autodesk.com/browse/DYN-4905)  I am not so sure that this is required. I will clarify at our next meeting

Comment (pinzart90) [reply src/DynamoCore/Configuration/PreferenceSettings.cs]: I removed the default trusted locations, as per our discussion

**Pull request merged**